include/ocvsmd/sdk/defines.hpp

```diff
@@ -8,13 +8,35 @@
 
 #include <cetl/pf20/cetlpf.hpp>
 
+#include <cerrno>
 #include <cstdint>
 
 namespace ocvsmd
 {
 namespace sdk
 {
 
+/// Defines some common error codes of IPC operations.
+///
+/// Maps to `errno` values, hence `std::int32_t` inheritance and zero on success.
+///
+enum class ErrorCode : std::int32_t  // NOLINT
+{
+    Success             = 0,
+    Busy                = EBUSY,
```

Comment:
Errno codes are platform-dependent. Since we're defining custom error codes anyway, it makes sense to assign well-defined platform-invariant constant values to each of them.

Reply:
We are defining only those enum cases for which we have specific business logic, BUT possible errno values are not limited by these cases. It could be anything. Also, my `ErrorCode` enum is NOT platform-invariant by design - requirements for ocvsmd were from day zero as Posix os. So, I don't see why we should complicate things (and IMHO make it really confusing comparing to quite minor "Success" issue) by trying make "universal"...

Reply:
BTW, I find `libcyphal`'s `PlatformError` stuff as really confusing, with all its complications (like unbounded_variant, interface, rtti - you name it). But ok, libcyphal is not bound to Posix, so maybe, maybe it worth to go with probably a bit of over engineering for the sake of platform-invariant-ness. My point - I don't feel right to replicate such approach at ocvsmd.